soc/arm/nxp_imx/rt117x/soc.c

```diff
@@ -0,0 +1,393 @@
+				clocks, clock_frequency));
+	CLOCK_EnableUsbhs0Clock(kCLOCK_Usb480M,
+				DT_PROP_BY_PHANDLE(DT_INST(0, nxp_kinetis_usbd),
+				clocks, clock_frequency));
+	USB_EhciPhyInit(kUSB_ControllerEhci0, CPU_XTAL_CLK_HZ, &usbPhyConfig);
+#endif
+
+#ifdef CONFIG_DISK_ACCESS_USDHC1
+	CLOCK_InitPfd(kCLOCK_PllSys2, kCLOCK_Pfd2, 24);
+	/* Configure USDHC clock source and divider */
+	rootCfg.mux = kCLOCK_USDHC1_ClockRoot_MuxSysPll2Pfd2;
+	rootCfg.div = 2;
+	CLOCK_SetRootClock(kCLOCK_Root_Usdhc1, &rootCfg);
+
+	/* ERR050396
+	 * Errata description:
+	 *  AXI to AHB conversion for CM7 AHBS port (port to access CM7 to
+	 * TCM) is by a NIC301 block, instead of XHB400 block. NIC301 doesn’t
+	 * support sparse write conversion. Any AXI to AHB conversion need
+	 * XHB400, not by NIC. This will result in data corruption in case
+	 * of AXI sparse write reaches the NIC301 ahead of AHBS.
+	 *
+	 * Errata workaround:
+	 *  For uSDHC, don’t set the bit#1 of IOMUXC_GPR28 (AXI transaction
+	 *  is cacheable), if write data to TCM aligned in 4 * bytes; No
+	 *  such write access limitation for OCRAM or external RAM
+	 */
+	IOMUXC_GPR->GPR28 &= (~IOMUXC_GPR_GPR28_AWCACHE_USDHC_MASK);
+#endif
+
+#ifdef CONFIG_VIDEO_MCUX_MIPI_CSI
+	/* MIPI CSI2 connect to CSI. */
+	CLOCK_EnableClock(kCLOCK_Video_Mux);
+	VIDEO_MUX->VID_MUX_CTRL.SET = (VIDEO_MUX_VID_MUX_CTRL_CSI_SEL_MASK);
+
+	rootCfg.mux = kCLOCK_CSI2_ClockRoot_MuxSysPll3Out;
+	rootCfg.div = 7;
+
+	CLOCK_SetRootClock(kCLOCK_Root_Csi2, &rootCfg);
+	CLOCK_SetRootClock(kCLOCK_Root_Csi2_Esc, &rootCfg);
+	CLOCK_SetRootClock(kCLOCK_Root_Csi2_Ui, &rootCfg);
+#endif
+
+#ifdef CONFIG_CAN_MCUX_FLEXCAN
+	rootCfg.mux = kCLOCK_CAN3_ClockRoot_MuxOsc24MOut;
+	rootCfg.div = 1;
+	CLOCK_SetRootClock(kCLOCK_Root_Can3, &rootCfg);
+#endif
+
+	GPC_CM_SetNextCpuMode(GPC_CPU_MODE_CTRL_0, kGPC_RunMode);
+	GPC_CM_SetNextCpuMode(GPC_CPU_MODE_CTRL_1, kGPC_RunMode);
+}
+
+#if defined(CONFIG_DISK_ACCESS_USDHC1) ||	\
+	defined(CONFIG_DISK_ACCESS_USDHC2)
+
+/* Usdhc driver needs to re-configure pinmux
+ * Pinmux depends on board design.
+ * From the perspective of Usdhc driver,
+ * it can't access board specific function.
+ * So SoC provides this for board to register
+ * its usdhc pinmux and for usdhc to access
+ * pinmux.
+ */
+
+static usdhc_pin_cfg_cb g_usdhc_pin_cfg_cb;
+
+void imxrt_usdhc_pinmux_cb_register(usdhc_pin_cfg_cb cb)
+{
+	g_usdhc_pin_cfg_cb = cb;
+}
+
+void imxrt_usdhc_pinmux(uint16_t nusdhc, bool init,
+	uint32_t speed, uint32_t strength)
+{
+	if (g_usdhc_pin_cfg_cb)
+		g_usdhc_pin_cfg_cb(nusdhc, init,
+			speed, strength);
+}
+#endif
+
+/**
+ *
+ * @brief Perform basic hardware initialization
+ *
+ * Initialize the interrupt controller device drivers.
+ * Also initialize the timer device driver, if required.
+ *
+ * @return 0
+ */
+
+static int imxrt_init(const struct device *arg)
+{
+	ARG_UNUSED(arg);
+
+	unsigned int oldLevel; /* old interrupt lock level */
+
+	/* disable interrupts */
+	oldLevel = irq_lock();
+
+#if __CORTEX_M == 7
```

Comment:
Should use `#ifdef CONFIG_SOC_MIMXRT1176_CM7` here

Reply:
Done.